suite/tests/CMakeLists.txt

```diff
@@ -3177,6 +3177,13 @@ if (CLIENT_INTERFACE)
       torunonly_drcachesim(delay-simple ${ci_shared_app}
         "-trace_after_instrs 20000 -exit_after_tracing 10000" "")
 
+      # We use a many-threaded test with a small max and test that we only see
+      # 1 thread, testing the thread ignore logic.  The max should be small enough
+      # to not be flaky on any platform.
```

Comment:
Not easily.  Maybe we could check that the instruction count plus load plus store is between 10K and 20K...though the current test setup of just having a regex makes that difficult, and it gets more in accurate for offline due to the compressed refs and post-processing expansion.  The runs on my machine have ~11K instructions which might be true across platforms: maybe we could check for that with a regex being `1....` (between 10K and 20K).  It feels like it might end up being flaky though.  WDYT?  Put in the instr count regex?

Reply:
I put in the regex.  We'll see if it turns the CI red now or at some point in the future.

Reply:
Maybe we can live with just testing whether the trace is indeed truncated.
A less flaky regex check would be to only verify # digits in truncated trace length < # digits in untruncated trace length.
For this we can set `-max_global_trace_refs` to a value << the original untruncated trace count.

Reply:
So the regex was a disaster: the test failed on all platforms due to variations in the count.  I'm reverting it.

Reply:
Nobody has the trace length for online though.  Maybe for offline the file size could be examined across two runs.  IMHO it's not worth all that effort: the single-traced-thread check already shows the truncation.

Reply:
In what I was suggesting, the "# digits in untruncated trace length" could be found by a single local run, and put in as a constant. 
But this is optional.

> the single-traced-thread check already shows the truncation

Yup, that should be sufficient.

Reply:
> In what I was suggesting, the "# digits in untruncated trace length" could be found by a single local run, and put in as a constant.

But that will vary based on compiler and shared libraries so I think that would end up having similar flakiness risks.  With a really complex regex or programmatic checking and the right threshold we could get a tighter bound (like not using 10K: some CI runs had 9K instrs and more data refs, so my simple regex failed) but it seems inherently fragile for these end-to-end tests.  Anyway I think we're in agreement; resolving.